src/main/java/repository/GeneralRepository.java

```diff
@@ -0,0 +1,223 @@
+package repository;
+
+
+import domain.Book;
+
+import exception.LoadException;
+import exception.SaveException;
+
+import java.io.*;
+import java.util.*;
+import java.util.stream.Collectors;
+
+import static domain.BookCondition.*;
+
+public class GeneralRepository implements Repository {
+
+    private static final String csvFileName = "/Users/kimnamgyu/desktop/study/dev-course/csvFile.csv";
+
+    @Override
+    public void load(List<Book> bookList) {
+        try (BufferedReader reader = new BufferedReader(new FileReader(csvFileName))) {
+            String readRecord;
+            while ((readRecord = reader.readLine()) != null) {
+                String[] record = readRecord.split(",");
+
+                int id = Integer.parseInt(record[0].trim());
+                String title = record[1].trim();
+                String author = record[2].trim();
+                int page = Integer.parseInt(record[3].trim());
+                String condition = record[4].trim();
+
+                // 도서를 가져오던 중 도서 정리중이면 대여 가능으로 변환
+                if (Objects.equals(condition, ORGANIZING.getCondition())) {
+                    condition = AVAILABLE.getCondition();
+                }
+
+                bookList.add(new Book(id, title, author, page, condition));
+            }
+        } catch (IOException e) {
+            throw new LoadException("CSV 파일을 읽어올 수 없습니다");
+        }
+    }
+
+    @Override
+    public void save(int id, String title, String author, int page, List<Book> bookList) {
+        try (BufferedWriter writer = new BufferedWriter(new FileWriter(csvFileName, true))) {
+            String newRecord = id + "," + title + "," + author + "," + page + "," + AVAILABLE.getCondition();
+
+            bookList.add(new Book(id, title, author, page, AVAILABLE.getCondition()));
+
+            writer.write(newRecord);
+            writer.newLine();
+        } catch (IOException e) {
+            throw new SaveException("CSV 파일에 저장할 수 없습니다.");
+        }
+    }
+
+    @Override
+    public List<Book> findByTitle(String searchTitle, List<Book> bookList) {
+        return bookList.stream()
+                .filter(book -> book.getTitle().toLowerCase().contains(searchTitle))
+                .map(book -> new Book(book.getId(), book.getTitle(), book.getAuthor(), book.getPage(), book.getCondition()))
+                .collect(Collectors.toList());
```

Comment:
```suggestion
return bookList.stream()
                .filter(book -> book.getTitle().toLowerCase().contains(searchTitle))
                .map(book -> new Book(book.getId(), book.getTitle(), book.getAuthor(), book.getPage(), book.getCondition()))
                .toList();
```

java16에서 toList()가 추가되었는데 한번 알아보시면 어떨까요?